ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.ts

```diff
@@ -33,120 +220,114 @@ export default {
     // --- State ---
     {
       id: 'UCU Firescorched Gain',
+      type: 'GainsEffect',
       netRegex: NetRegexes.gainsEffect({ effectId: '1D0' }),
       condition: Conditions.targetIsYou(),
       run: (data) => data.fireDebuff = true,
     },
     {
       id: 'UCU Firescorched Lose',
+      type: 'LosesEffect',
       netRegex: NetRegexes.losesEffect({ effectId: '1D0' }),
       condition: Conditions.targetIsYou(),
       run: (data) => data.fireDebuff = false,
     },
     {
       id: 'UCU Icebitten Gain',
+      type: 'GainsEffect',
       netRegex: NetRegexes.gainsEffect({ effectId: '1D1' }),
       condition: Conditions.targetIsYou(),
       run: (data) => data.iceDebuff = true,
     },
     {
       id: 'UCU Icebitten Lose',
+      type: 'LosesEffect',
       netRegex: NetRegexes.losesEffect({ effectId: '1D1' }),
       condition: Conditions.targetIsYou(),
       run: (data) => data.iceDebuff = false,
     },
     {
       id: 'UCU Fireball Counter',
+      type: 'Ability',
       netRegex: NetRegexes.ability({ id: '26C5', source: 'Firehorn' }),
       netRegexDe: NetRegexes.ability({ id: '26C5', source: 'Feuerhorn' }),
       netRegexFr: NetRegexes.ability({ id: '26C5', source: 'Corne-De-Feu' }),
       netRegexJa: NetRegexes.ability({ id: '26C5', source: 'ファイアホーン' }),
       netRegexCn: NetRegexes.ability({ id: '26C5', source: '火角' }),
       netRegexKo: NetRegexes.ability({ id: '26C5', source: '화염뿔' }),
       run: (data, matches) => {
-        data.fireballs[data.naelFireballCount].push(matches.target);
+        (data.fireballs[data.naelFireballCount] ??= []).push(matches.target);
```

Comment:
Is this upset about naelFireballCount being 0 and 0 not being initialized? Or will it just always be mad since it has no idea what the indexes are aside from being a number?

Reply:
Yeah, `data.fireballs[number]` is potentially undefined always.